### PR TITLE
Move syntax highlighting to web worker

### PR DESCRIPTION
**Pull request by silverwind**
This should eliminate page freezes when loading big files/diff. `highlightBlock` is needed to preserve our existing HTML nodes when highlighting and for that, highlight.js needs access to the DOM API so I added a DOM implementation to make it work, which adds around [300kB](https://bundlephobia.com/result?p=domino@2.1.4) to the output file size of the lazy-loaded `highlight.js` chunk.

### CODE REVIEW

**Comment by silverwind** [comment]
It might be worth removing `RequireHighlightJS` from template context and let the decision be done in JS alone. I'm just not sure if it's going to break anything like [here](https://github.com/go-gitea/gitea/blob/08208cdfb87bc3f0b534eda9d9ef265dcfbc18c3/routers/repo/blame.go#L113-L114).

File: web_src/js/index.js [hunk 1]
```diff
@@ -2628,9 +2627,11 @@ $(document).ready(async () => {
     }
   });
 
+  const codeBlocks = document.querySelectorAll('pre code');
+
   // parallel init of lazy-loaded features
-  [hljs] = await Promise.all([
-    initHighlight(),
+  await Promise.all([
+    (codeBlocks && codeBlocks.length) && highlight(codeBlocks),
```
Comment (guillep2k) [inline web_src/js/index.js]: Double check (here and in `highlight`) seems unnecessary?

Comment (silverwind) [reply web_src/js/index.js]: Indeed. Guess I was getting a bit overzealous with those safeguards.

**Comment by silverwind** [comment]
Regardin removal of `.RequireHighlightJS`: I guess it cannot be done without breaking the example [here](https://github.com/go-gitea/gitea/blob/master/docs/content/doc/advanced/customizing-gitea.en-us.md#example-mermaidjs) which seems to piggy-back on the variable for detection of rendered Markdown.

**Comment by silverwind** [comment]
Fixed on race condition that happened with multiple outstanding highlight requests by adding a `index` to the data submitted to the worker. This should work reliably as every call to `highlight` will instantiate their own worker (the chunk is still only downloaded once).

File: web_src/js/features/highlight.js [hunk 2]
```diff
@@ -1,12 +1,23 @@
...
+    const {index, html} = data;
+    nodes[index].outerHTML = html;
   }
 
-  return hljs;
+  for (let index = 0; index < nodes.length; index++) {
+    const node = nodes[index];
+    if (!node) continue;
+    const html = node.outerHTML;
+    if (!html) continue;
```
Comment (guillep2k) [inline web_src/js/features/highlight.js]: How come a node can _not have_ an `outerHTML`? 🤔

Comment (silverwind) [reply web_src/js/features/highlight.js]: Yeah, not possible, removing

**Review by lafriks — APPROVED** [review]

**Review by guillep2k — APPROVED** [review]

**Comment by lafriks** [comment]
Make lgtm work

**Pull request merged**